fix: rename rule no-empty-enum-servers to no-server-variables-empty-enum

adamaltman opened the pull request:
## What/Why/How?

- rename `no-empty-enum-servers` to `no-server-variables-empty-enum` and adjust corresponding code and docs. The current live docs are completely incorrect and wouldn't function, so it is safer to rename (the actual working one would have been `no-servers-empty-enum` but it wasn't documented that way due to the strange naming in the source code: https://redocly.com/docs/cli/rules/no-empty-enum-servers/ -- the changelog should note the rename.
- docs: fix `no-server-example.com` docs

## Reference

- Fixes #812 

## Testing

Passed.

## Check yourself

- [x] Code is linted
- [ ] **N/A** Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

RomanHotsiy reviewed (approved):
This looks good.

The only concert is it is it technically a breaking change. Someone could have those rules ignored.

But I think this is okay.

adamaltman commented:
Are we sure we don't want to add a commit to the CHANGELOG in an unreleased section at the top? Anyway, here is what I propose for simplicity during the release.

Changes
- Renamed `no-servers-empty-enum` to `no-server-variables-empty-enum` and fixed incorrect docs of `no-empty-enum-servers`.

Fixes
- Fixed docs for no-server-example.com
